python-package/lightgbm/basic.py

```diff
@@ -58,9 +58,9 @@ def register_logger(logger):
     _LOGGER = logger
 
 
-def _normalize_native_string(func):
+def _normalize_native_string(func: Callable[[str], None]) -> Callable[[str], None]:
     """Join log messages from native library which come by chunks."""
-    msg_normalized = []
+    msg_normalized: List[str] = []
 
     @wraps(func)
     def wrapper(msg):
```

Comment:
Was `wrapper()` simply missed or you haven't annotated it due to problems with `Union` in return type?
https://github.com/python/mypy/issues/1693

Reply:
oh just missed

Reply:
updated in https://github.com/microsoft/LightGBM/pull/4527/commits/a1bec4ed1b1cd99290996dd7f4435855c14abb37, `mypy` didn't have any complaints so I think it's ok